src/contextProviders/Web3Provider.jsx

```diff
@@ -77,34 +69,21 @@ const Web3Provider = props => {
     onboard.walletCheck().then();
   };
 
-  const displayForeignNetRequiredWarning = (onClose, buttonLabel = 'Close') => {
-    setForeignNetRequiredWarning({
-      showForeignNetRequiredWarning: true,
-      onForeignNetWarningClose: onClose,
-      foreignNetWarningButtonLabel: buttonLabel,
-    });
-  };
+  const { networkId, isForeignNetwork, isHomeNetwork } = network;
+  const isMetaMask = web3 && web3.MetaMask;
+  const isEnabled = !!web3 && !!account && !!balance && !!networkId;
 
-  const displayHomeNetRequiredWarning = (onClose, buttonLabel = 'Close') => {
-    setHomeNetRequiredWarning({
-      showHomeNetRequiredWarning: true,
-      onHomeNetWarningClose: onClose,
-      homeNetWarningButtonLabel: buttonLabel,
-    });
+  const switchNetwork = () => {
+    if (isMetaMask) {
+      window.ethereum.request({
+        method: 'wallet_switchEthereumChain',
+        params: [
+          { chainId: isForeignNetwork ? config.homeNetworkChainId : config.foreignNetworkChainId },
```

Comment:
This implementation has issue when user wallet is connected to a third network (not home or foreign) and tries to donates! In this case the wallet will try to connect to foreign network instead of home network.

I suggest changing `ActionNetworkWarning` to accept an enum type parameter which indicates the aimed network